### PR TITLE
#152220230 Implement user signup

### PR DESCRIPTION
**Pull request by tomipaul**
#### What does this PR do?
Implement user signup

#### Description of Task to be completed?
- Create utility functions, controllers and middleware
- Define route and mount controllers/middleware

#### What are the relevant pivotal tracker stories?
#152220230

#### Screenshots (if appropriate)
nil

### CODE REVIEW

**Comment by tomipaul** [comment]
@supermike, @3m3kalionel @ayodelevm please review

File: src/server/controllers/AuthController.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      return getUser({ username })
+        .then(user =>
+          user.verifyPassword(password)
+            .then((passwordIsValid) => {
+              if (passwordIsValid) {
+                const rsaKey = process.env.PRIVATE_KEY;
+                return generateToken(user, rsaKey);
+              }
+              const error = new Error('Invalid Password');
+              error.code = 422;
```
Comment (mikey2020) [inline src/server/controllers/AuthController.js]: Am curious to why you decided to return a `422` when you have an invalid password

Comment (mikey2020) [reply src/server/controllers/AuthController.js]: You  used `err` before but now you are using `error`, you should try and be consistent when defining variables

**Comment by tomipaul** [comment]
@mikey2020 please affirm that the suggested corrections have been made

**Pull request merged**